In [1]:
import plotly.express as px 
import pandas as pd 
import numpy as np
import matplotlib.pyplot as plt

## Ex 1: 

### Data

In [2]:
x=[1,3,2,3,2,2,3,1,2,3,2,3,1,2,3,4,1,2,2,2,2,3,4,1,2,3,4,1,4,3,2,1,4,2,4,3,1,3,2,2]
y=[6,2,4,1,4,1,3,5,2,2,5,2,6,6,2,2,5,5,1,1,4,2,1,5,4,2,1,5,4,3,2,1,1,6,2,1,4,2,3,5]
df = pd.DataFrame({'x': x, 
                   'y': y})
df.head()

x  y
0  1  6
1  3  2
2  2  4
3  3  1
4  2  4

In [3]:
len(df.value_counts()) 

16

In [4]:
df.describe()

x          y
count  40.000000  40.000000
mean    2.375000   3.075000
std     0.978945   1.730385
min     1.000000   1.000000
25%     2.000000   2.000000
50%     2.000000   2.500000
75%     3.000000   5.000000
max     4.000000   6.000000

In [5]:
fig = px.scatter(x=df['x'], y=df['y'])
fig.show()

Questao (A)

In [11]:
df_filtred = df[(df['x'] > np.median(x)) & (df['y'] > np.median(y))] 
print('ACIMA DA MEDIA')
df_filtred

x  y
6   3  3
28  4  4
29  3  3

In [12]:
df_filtred = df[(df['x'] < np.median(x)) & (df['y'] < np.median(y))] 
print('ABAIXO DA MEDIA')
df_filtred

BAIXO


x  y
31  1  1

Questao (B)

In [18]:
filtered_count = len(df[(df['x'] < np.median(x)) & (df['y'] < np.median(y))]) 
percent = filtered_count/len(df)*100
print(f'--> {percent}%')

--> 2.5%


Questao (C)

In [31]:
filtered_count = len(df[df['y'] < np.median(y)]) 
percent = filtered_count/len(df)*100
print(f'--> {percent}%')

--> 50.0%


Questao (D)

In [39]:
ganham_pouco_e_baixa_rotatividade_count = len(df[(df['x'] < np.median(x)) & (df['y'] < np.median(y))]) 
baixa_rotatividade_count = len(df[df['x'] < np.median(x)])
percent = ganham_pouco_e_baixa_rotatividade_count/baixa_rotatividade_count*100
print(f'Ganham pouco e tem baixa rotatividade = {ganham_pouco_e_baixa_rotatividade_count} | Baixa Rotatividade = {baixa_rotatividade_count}')
print(f'--> {percent}%')

Ganham pouco e tem baixa rotatividade = 1 | Baixa Rotatividade = 8
--> 12.5%


Questao (E)

Sim, signigfica que as pessoas que possuem baixa rotatividade de emprego tendem a ganhar mais que a media. Uma vez que apensa 12.5% das pessoas que possuem 

Questao (F)

In [96]:
filter_duplicates = df.drop_duplicates(ignore_index=True).rename(columns={'x': 'Rotatividade de Emprego', 'y':'Salarios Minimos'})
filter_duplicates = filter_duplicates.sort_values('Salarios Minimos', ascending=False, ignore_index=True)
filter_duplicates

Rotatividade de Emprego  Salarios Minimos
0                         1                 6
1                         2                 6
2                         2                 5
3                         1                 5
4                         4                 4
5                         1                 4
6                         2                 4
7                         3                 3
8                         2                 3
9                         4                 2
10                        3                 2
11                        2                 2
12                        2                 1
13                        3                 1
14                        4                 1
15                        1                 1

In [97]:
fig = px.bar(filter_duplicates)
fig.show()